### ISSUE TITLE
Error: uninitialized constant ActionMailer

### ISSUE THREAD

**Issue by muresanbeniamin** [cucumber/cucumber-rails#483]
After updating from 2.0.0 to 2.1.0 I get this error message when running the specs.

```
Before hook at vendor/bundle/ruby/2.6.0/gems/cucumber-rails-2.1.0/lib/cucumber/rails/hooks/mail.rb:3

Message:

    uninitialized constant ActionMailer
Did you mean?  ActionView (NameError)
```

**Comment by luke-hill** [comment]
Do you mean v2.1.0?
Do you use ActiveRecord.

A couple of bugs are currently fixed in master. I will cut a release soon, that affects people using rails but not AR

**Comment by muresanbeniamin** [comment]
> Do you mean v2.1.0? Do you use ActiveRecord?
> 
> A couple of bugs are currently fixed in master. I will cut a release soon, that affects people using rails but not AR

Yes, ActiveRecord is used and yes, I meant v.2.1.0 - updated description too.
Alright then, thanks for now.

**Comment by luke-hill** [comment]
If you're using AR then you **shouldn't** hit these bugs.

Can you follow the template and provide something reproducible.

**Comment by ykitamura-mdsol** [comment]
I'm facing the same issue. Looks like this change is affecting: https://github.com/cucumber/cucumber-rails/pull/441/files#diff-d09a3470ddd8f22434f9777cfdaf3c0f

So if you don't require `action_mailer` in "config/application.rb":
```ruby
# Pick the frameworks you want:
require "active_model/railtie"
 :
# require "action_mailer/railtie"
```

you will get the error. 

Maybe it's good to bring back `defined?(ActionMailer::Base)` check.

**Comment by luke-hill** [comment]
This is a good point. But if you're using all the rails standards, then you **shouldn't** hit this error.

To my knowledge some of these were reverted a bit ago due to people having issues. The person who reverted some said at the time some could remain as forced and some optional. Making them optional again isn't a big issue. I'm just trying to understand if something else is going on here.

If the OP can paste us maybe his gemfile / stack it would help triage.

**Comment by Draiken** [comment]
ActionMailer dependency should be optional (just like it is on Rails).
We're also hitting this issue, since our app doesn't use ActionMailer.

Could we wrap the hook in an `if defined?(ActionMailer)` conditional?

**Comment by luke-hill** [comment]
Fixed in master. Will be released in v2.2.0 available from tomorrow.
